### ISSUE TITLE
Newsletter Categories API: Create GET endpoint to retrieve the list of newsletter categories.

### ISSUE THREAD

**Issue by phcp** [Automattic/wp-calypso#80036]
This task aims to create the GET endpoint to retrieve the list of newsletter categories. The data returned will be used on the Subscribe Block and possibly on the Categories Management to identify whether a category is a newsletter category or not.

More details about the data structure can be found here: pdDOJh-26F-p2

- [x] Define response schema
- [x] Create V2 GET endpoint
- [x] Create automated tests

**Comment by phcp** [comment]
Endpoint: `GET /wpcom/v2/sites/<siteid>/newsletter-categories`

Response Schema:

```json
{
    "newsletter_categories": [
        {
            "id": 1,
            "name": "Uncategorized",
            "slug": "uncategorized",
            "description": "",
            "parent": 0
        }
    ]
}
```

**Comment by TimBroddin** [comment]
Shouldn't we create those under the `/wpcom/v2` namespace? I think `/wp/v2` is for core WP endpoints.

We had some discussion about endpoints on p1690876521361669-slack-C02TCEHP3HA, which I summarized over [here](https://github.com/Automattic/wp-calypso/issues/80026#issuecomment-1660099748)

**Comment by phcp** [comment]
> Shouldn't we create those under the /wpcom/v2 namespace? I think /wp/v2 is for core WP endpoints.

Yes, indeed. Thanks for catching that. I was just writing down some rough notes, nothing implemented yet.

**Comment by phcp** [comment]
Closed by D117699-code
